File: Challenges/450_DSA_Cracker/017_Best_Time_To_Buy_Sell_2.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profits = 0
        n = len(prices)
        for i in range(0, n-1) : 
            if prices[i] < prices[i+1] : 
                profits += prices[i+1] - prices[i]
        return profits

File: Challenges/450_DSA_Cracker/Array/017_Best_Time_To_Buy_Sell_2.java
class Solution {
    public int maxProfit(int[] prices) {
        int profits = 0;
        for(int i=0; i<prices.length-1; i++) {
            if(prices[i] < prices[i+1])
                profits += prices[i+1] - prices[i];
        }
        return profits;
    }
}
